src/services/blockService.ts

```diff
@@ -0,0 +1,103 @@
+import { DataAccessLayer } from "@blockr/blockr-data-access";
+import { Block } from "@blockr/blockr-models";
+import { NextFunction } from "connect";
+import { Request, Response } from "express";
+import { inject, injectable } from "inversify";
+import { HttpException } from "../utils/exceptions/httpException";
+import logger from "../utils/logger";
+
+@injectable()
+export class BlockService {
+    protected dataAccessLayer: DataAccessLayer;
+
+    constructor(@inject(DataAccessLayer) dataAccessLayer: DataAccessLayer) {
+        this.dataAccessLayer = dataAccessLayer;
+    }
+
+    public async getBlocksAsync(request: Request, response: Response, next: NextFunction) {
+        try {
+            response.send(
+                request.query.number !== undefined ? await this.getBlockByNumberAsync(request.query.number)
```

Comment:
Ternary statements are in general better then if statements. However, I'm not sure how I feel about a quadruple nested ternary statement.